### PR TITLE
Removed most s3 bucket based tests (replaced with UC Volumes)

### PR DESCRIPTION
**Pull request by rithwik-db**
# What does this PR do?

Essentially, all usages of `s3_bucket` has been removed outside of `test_object_store.py` and `test_s3_object_store.py` since we have the ephemeral and read_only paths in UC Volumes now.

### CODE REVIEW

File: tests/trainer/test_fsdp_checkpoint.py [hunk 1]
```diff
@@ -971,19 +970,18 @@ def test_elastic_resumption(
...
     if autoresume:
         run_name = 'my-autoresume-run'
     else:
         run_name = None
 
-    base_path = (
-        f's3://{s3_bucket}/{s3_read_only_prefix}/elastic_test/'
-        f'{sharding_strategy.lower()}_sharded_{precision}_'
-        f'{num_shards}/'
+    base_path = os.path.join(
```
Comment (rithwik-db) [inline tests/trainer/test_fsdp_checkpoint.py:981]: Same issue with `.metadata` not existing (daily tests that fail: https://github.com/mosaicml/composer/actions/runs/15356296375/job/43215949886) cc: @ethantang-db, can you lmk when you move this files? Thanks!

**Comment by dakinggg** [comment]
Ah I think its because dist is initialized with gpu (i.e. nccl) rather than cpu (i.e gloo) when running the gpu tests. its probably not worth adjusting test set up to support this, and just let the test run on gpu instead of cpu

File: composer/utils/reproducibility.py [hunk 2]
```diff
@@ -232,6 +232,16 @@ def load_rng_state(rng_state_dicts: list[dict[str, Any]]):
...
+            # PyTorch 2.6 strictly enforces the size of the state dict values vs the size in the checkpoint.
+            # PyTorch 2.1 has moved to a RNG of size 16 instead of 816
+            # Therefore, errors occur if we load a ckpt at or after 2.6 if the ckpt was saved before 2.1.
+            # To avoid this, we zero out the CUDA RNG state if the size is incorrect.
+            if rng_state_dict['cuda'].shape[0] != 16:
+                warnings.warn(
+                    'The CUDA RNG state was saved with a different version of PyTorch than the current version.'
+                    'CUDA RNG state will be zeroed out instead.',
+                )
+                rng_state_dict['cuda'] = torch.zeros(16, dtype=torch.uint8)
```
Comment (rithwik-db) [inline composer/utils/reproducibility.py:244]: It seems that since we updated to use gpu, the `rng_state_dict` is relevant now and this is causing issues with equality in the metrics/models. I reduced the tolerance to get the tests to pass on the daily workflow but the current tolerances work fine on my remote instance. Not sure what I am missing...

**Comment by dakinggg** [comment]
I'm ok just adjusting the tolerance for 2.6 if thats sufficient for the test to pass. Not super critical to keep exact numerical determinism across torch versions - and in fact on GPU I'd guess this might not be possible.

File: tests/trainer/test_checkpoint.py [hunk 3]
```diff
@@ -1154,42 +1155,49 @@ def test_save_ignore_keys(self, save_ignore_keys, weights_equal, callbacks_equal
...
     @pytest.mark.parametrize(
-        'remote_checkpoint_uri, remote_checkpoint_name, continue_training_dur, final_checkpoint_name',
+        'remote_checkpoint_name, continue_training_dur, final_checkpoint_name',
         [
-            ['backwards_compatibility/trained_ckpt_cpu_ep2.pt', 'ep2.pt', '3ep', 'ep3.pt'],
+            ['ep2.pt', '3ep', 'ep3.pt'],
         ],
     )
     @pytest.mark.filterwarnings('ignore:.*The checkpoint included CUDA RNG state.*')
+    @pytest.mark.filterwarnings('ignore:.*The CUDA RNG state could not be loaded from the checkpoint.*')
```
Comment (dakinggg) [inline tests/trainer/test_checkpoint.py:1167]: don't need to filter this anymore, right?

File: tests/trainer/test_fsdp_checkpoint.py [hunk 4]
```diff
@@ -525,13 +525,14 @@ def test_fsdp_mixed_with_sync(
 @pytest.mark.filterwarnings(r'ignore:.*The CUDA RNG state could not be loaded.*:UserWarning')
 @pytest.mark.filterwarnings(r'ignore:.*ShardedTensor.to only move tensor to its current device.*:UserWarning')
 @pytest.mark.filterwarnings(r'ignore:.*\(TP\) is experimental.*:FutureWarning')
+@pytest.mark.filterwarnings(r'ignore:.*The CUDA RNG state was saved with a different version of PyTorch.*')
```
Comment (dakinggg) [inline tests/trainer/test_fsdp_checkpoint.py:528]: why was this change needed? we didn't change the checkpoints being used here right?

**Review by dakinggg — APPROVED** [review]

**Comment by rithwik-db** [comment]
Reran daily tests to make sure with latest fixes: https://github.com/mosaicml/composer/actions/runs/15500553934
